components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/IdentityGovernanceServiceImpl.java

```diff
@@ -233,10 +239,34 @@ public ConnectorConfig getConnectorWithConfigs(String tenantDomain,
 
         for (ConnectorConfig connectorConfig : connectorListWithConfigs) {
             if (connectorConfig.getName().equals(connectorName)) {
+                // Should remove this logic eventually.
+                migratePropertyUseNumericToUseAlphaNumeric(connectorName, connectorConfig);
                 return connectorConfig;
             }
         }
         return null;
     }
 
+    /**
+     * This method is used to migrate the property value useNumericCharacters to useAlphanumericCharacters.
+     *
+     * @param connectorName   Name of the connector.
+     * @param connectorConfig Connector configuration.
+     */
+    private void migratePropertyUseNumericToUseAlphaNumeric(String connectorName, ConnectorConfig connectorConfig) {
+
+        if (connectorName.equals("email-otp-authenticator")) {
+            if (connectorConfig.getProperties().length == 6) {
```

Comment:
Why are we hardcoding these values?

Reply:
The constant cannot be imported from the email-authenticator, as it creates a cyclic dependency between the modules. As this is a temporarily added logic, we will be removing the script after eventual migration.